src/AgentCommerce/Util/PayPalCartFactory.php

```diff
@@ -0,0 +1,96 @@
+<?php declare(strict_types=1);
+/*
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Swag\PayPal\AgentCommerce\Util;
+
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\PayPalSDK\Struct\AgenticCommerce\V1\Address;
+use Shopware\PayPalSDK\Struct\AgenticCommerce\V1\CartItemCollection;
+use Shopware\PayPalSDK\Struct\AgenticCommerce\V1\PayPalCart;
+use Swag\PayPal\AgentCommerce\Exception\AgentException;
+
+/**
+ * @internal
+ */
+#[Package('checkout')]
+class PayPalCartFactory
+{
+    public function create(array $data): PayPalCart
+    {
+        $payPalCart = (new PayPalCart())->assign($data);
+
+        $this->validateCustomerData($payPalCart);
+
+        if (!$payPalCart->isset('items') || !$payPalCart->getItems()->count()) {
+            throw AgentException::requiredFieldsMissing('items');
+        }
+
+        $this->validateItems($payPalCart->getItems());
+
+        return $payPalCart;
+    }
+
+    private function validateCustomerData(PayPalCart $cart): void
+    {
+        $customer = $cart->getCustomer();
+        if (!$customer) {
+            return;
+        }
+
+        if (!$customer->getEmailAddress()) {
+            throw AgentException::requiredFieldsMissing('cart.customer.emailAddress');
+        }
+
+        if (!$customer->isset('name') || !$customer->getName()->isset('givenName') || !$customer->getName()->isset('surname')) {
+            throw AgentException::requiredFieldsMissing('cart.customer.name');
+        }
+
+        $shippingAddress = $cart->getShippingAddress();
+        if (!$shippingAddress instanceof Address) {
+            throw AgentException::requiredFieldsMissing('cart.shippingAddress');
+        }
+
+        $this->validateAddress($shippingAddress);
+
+        if ($cart->getBillingAddress()) {
+            $this->validateAddress($cart->getBillingAddress());
+        }
+    }
+
+    private function validateAddress(Address $address): void
+    {
+        if (!$address->getAddressLine1()) {
+            throw AgentException::requiredFieldsMissing('addressLine1');
+        }
+
+        if (!$address->getAddressLine2()) {
+            throw AgentException::requiredFieldsMissing('addressLine2');
+        }
+
+        if (!$address->getCountryCode()) {
+            throw AgentException::requiredFieldsMissing('countryCode');
+        }
+    }
+
+    private function validateItems(CartItemCollection $items): void
+    {
+        foreach ($items as $key => $item) {
+            if (!$item->isset('variantId')) {
+                throw AgentException::requiredFieldsMissing(\sprintf('cart.items.%s.variantId', $key));
+            }
+
+            if (!Uuid::isValid($item->getVariantId() ?? '')) {
+                throw AgentException::requiredFieldsMissing(\sprintf('cart.items.%s.variantId  not valid uuid', $key));
```

Comment:
Either just say it's missing or add a new exception to explain whats up, as "Required field 'cart.items.%s.variantId not valid uuid' is missing" is not a good exception message:
```suggestion
                throw AgentException::requiredFieldInvalid(\sprintf('cart.items.%s.variantId', $key), 'not a valid uuid');
```